2016/proposals/rai-vivek.md

```diff
@@ -180,28 +185,49 @@ systems.
-1  | ugc@edu.in        | 'Linux'  | 'ghostbook'  | '3.19.0-47-generic' | '#53~14.04.1-Ubuntu SMP Mon Jan 18 16:09:14 UTC 2016’ | 'x86_64' | 'x86_64'  | 'foo@bar.com' | 'Tue Mar 8 00:52:35 2016'  | 0
-2  | trainus@gmail.com | 'Darwin' | 'Vs-MacBook' | '15.0.0'            | 'Darwin Kernel Version 15.0.0: ..  ’                  | 'x86_64' | 'i386'    | 'bar@foo.com' | 'Sat Mar 12 18:34:12 2016' | 0
+id | workshop_id         | system   | hostname     | release             | version                                               | machine  | processor | email         | time                       | exit_status
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+1  | 'ugc@edu.in'        | 'Linux'  | 'ghostbook'  | '3.19.0-47-generic' | '#53~14.04.1-Ubuntu SMP Mon Jan 18 16:09:14 UTC 2016’ | 'x86_64' | 'x86_64'  | 'foo@bar.com' | 'Tue Mar 8 00:52:35 2016'  | 0
+2  | 'trainus@gmail.com' | 'Darwin' | 'Vs-MacBook' | '15.0.0'            | 'Darwin Kernel Version 15.0.0: ..  ’                  | 'x86_64' | 'i386'    | 'bar@foo.com' | 'Sat Mar 12 18:34:12 2016' | 0
 
 (TABLE software_version)
 
-id | machine_id | software | installed | version | log
-------------------------------------------------------
-1  | 1          | python   | True      | 3.5.1   | ''
-2  | 1          | git      | True      | 1.9.1   | ''
-3  | 1          | hg       | False     | ''      | ''
-4  | 2          | pip      | True      | 7.9     | 'A newer version of pip is available.'
+id | machine_id | software   | installed | version  | log
+---------------------------------------------------------
+1  | 1          | 'python'   | True      | '3.5.1'  | ''
+2  | 1          | 'git'      | True      | '1.9.1'  | ''
+3  | 1          | 'hg'       | False     | ''       | ''
+4  | 2          | 'pip'      | True      | '7.9'    | 'A newer version of pip is available.'
 ```
 
 Once the model is defined, queries [complex
 queries](http://flask-sqlalchemy.pocoo.org/2.1/queries/) can be performed to
 add, remove or update records in the database as per requirement.
 
+A few examples of schema models are shown below:
+
+```python
+>>> pip = PackageInfo(package='pip', installed=1, version='7.9', log='Upgrade', machine_id=1)
+
+>>> pip
+<PackageInfo(id=1, software=pip, installed=True, version=7.9, log=Upgrade, machine_id=1)>
+
+>>> # add more packages, machines
+
+>>> MachineInfo.query.all()
+[<MachineInfo(id=1, workshopid="ugc@edu.net", system=Linux, ..., exit_status=0)>,
+ <MachineInfo(id=2, workshopid="example@ready.net", system=Darwin, ..., exit_status=0)>]
+
+>>> MachineInfo.query.first().packages
+[<PackageInfo(id=1, software=git, installed=True, version=1.9, log=, machine_id=1)>,
+ <PackageInfo(id=2, software=pip, installed=True, version=7.9, log=Upgrade, machine_id=1)>]
+
+```
+
```

Comment:
So regarding https://github.com/numfocus/gsoc/pull/94#issuecomment-198775378: you can replace the table below "A typical example of the database entry then may look like" with these examples.